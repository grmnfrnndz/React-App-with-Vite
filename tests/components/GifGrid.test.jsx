
import React from 'react';
import {render, screen} from "@testing-library/react";
import {GifGrid} from "../../src/components/index.js";


jest.mock("../../src/hooks/useFetchGifs");



describe('test on GifGrid', function () {

    const category = 'Saitama';

    test('match snaspshot', () => {
        const {container} = render(<GifGrid category={category}/>);
        expect(container).toMatchSnapshot();
    });

    test('show loading', () => {
        UseFetchGifs.mockReturnValue({
            images: [],
            isLoading: true,
        });


        render(<GifGrid category={category} />);

        expect(screen.getByText('Cargando...'));
        expect(screen.getByText(category));

        screen.debug();
    });

    test('show item when load imagen from UseFetchGifs', () => {
        render(<GifGrid category={category} />);
    });

});