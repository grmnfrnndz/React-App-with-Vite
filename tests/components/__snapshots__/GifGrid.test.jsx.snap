// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on GifGrid match snaspshot 1`] = `
<div>
  <h3>
    Saitama
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
